src/main/java/org/jvnet/hudson/test/TemporaryDirectoryAllocator.java

```diff
@@ -50,8 +50,15 @@ public TemporaryDirectoryAllocator(File base) {
     }
 
     public TemporaryDirectoryAllocator() {
-        this.base = new File(System.getProperty("java.io.tmpdir"), "jenkinsTests.tmp");
-        base.mkdirs();
+        File tempDir = new File(System.getProperty("java.io.tmpdir"), "jenkinsTests.tmp");
+        if (tempDir.mkdirs() == false) {
+            try {
+                File tempDir2 = java.nio.file.Files.createTempDirectory("jenkinsTests").toFile();
```

Comment:
Better to use `File.createTempFile` so we can run on Java 6, since currently this library uses `-target 6`. It simply forgot to use Animal Sniffer.

I am fine with updating to 7 or even 8 (does not really harm even use from plugins with old core baselines), but if we are going to do that, we should change the source level and introduce Animal Sniffer at once.